pkg/service/providerindex/legacy.go

```diff
@@ -51,7 +61,7 @@ func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Mult
 			return nil, err
```

Comment:
🤷‍♂️ yep, not strong feelings on this. It shouldn't happen.

The assumption you're making here is that the 3 claims together are required by the caller, but they could be 3 location claims for the same hash and returning just 1 of them would allow the data to be fetched. Personally I think it's better for the service to not be smart, and just return what it can - the client needs to be smart about what it uses from the results regardless, and should to be able to deal with imperfect/incomplete data.

Reply:
as discussed sync, the provider index service will now attempt to fetch results from IPNI before trying to find them in legacy claims, rather than the other way around. Thus, fetching claims from legacy claims is a best effort to find relevant claims, and the legacy claims service will return whatever it is able to find, even if that's partial results.